observation/reference_pointing.py

```diff
@@ -463,72 +42,30 @@ def save_pointing_offsets(session, pointing_offsets, middle_time):
-        target.add_tags('bfcal single_accumulation')
-        session.label('interferometric_pointing')
-        user_logger.info("Initiating interferometric pointing scan on target "
-                         "'%s' (%d pointings of %g seconds each)",
-                         target.name, len(offsets), opts.track_duration)
-        session.track(target, duration=0, announce=False)
-        # Point to the requested offsets and collect extra data at middle time
-        for n, offset in enumerate(offsets):
-            user_logger.info("initiating track on offset of (%g, %g) degrees", *offset)
-            session.ants.req.offset_fixed(offset[0], offset[1], opts.projection)
-            session.track(target, duration=opts.track_duration, announce=False)
-            offset_end_times[n] = time.time()
-            if not kat.dry_run and n == len(offsets) // 2 - 1:
-                # Get weather data for refraction correction at middle time
-                temperature = kat.sensor.anc_air_temperature.get_value()
-                pressure = kat.sensor.anc_air_pressure.get_value()
-                humidity = kat.sensor.anc_air_relative_humidity.get_value()
-                weather = {'temperature': temperature, 'pressure': pressure,
-                           'humidity': humidity}
-                middle_time = offset_end_times[n]
-                user_logger.info("reference time = %.1f, weather = "
-                                 "%.1f deg C | %.1f hPa | %.1f %%",
-                                 middle_time, temperature, pressure, humidity)
-        # Clear offsets in order to jiggle cal pipeline to drop its final gains
-        # XXX We assume that the final entry in `offsets` is not the origin
-        user_logger.info("returning to target to complete the scan")
-        session.ants.req.offset_fixed(0., 0., opts.projection)
-        session.track(target, duration=0, announce=False)
-        user_logger.info("Waiting for gains to materialise in cal pipeline")
-
-        # Perform basic interferometric pointing reduction
-        if not kat.dry_run:
-            # Wait for last piece of the cal puzzle (crash if not on time)
-            last_offset_start = offset_end_times[-1] - opts.track_duration
-            session.get_cal_solutions('G', timeout=300.,
-                                      start_time=last_offset_start)
-            user_logger.info('Retrieving gains, fitting beams, storing offsets')
-            data_points = get_offset_gains(session, offsets, offset_end_times,
-                                           opts.track_duration)
-            beams = fit_primary_beams(session, data_points)
-            pointing_offsets = calc_pointing_offsets(session, beams, target,
-                                                     middle_time, **weather)
-            save_pointing_offsets(session, pointing_offsets, middle_time)
+        target.add_tags("bfcal single_accumulation")
+        session.reference_pointing_scan(
```

Comment:
At this stage, please include a normal 'session.track' call, followed by the 'session.reference_pointing_scan' call, followed by another 'session.track' call. Use the same target for all 3 calls for now, but I think we will eventually choose a different reference pointing target which Ludwig can maybe help us with.